lib/xml/parser.js

```diff
@@ -82,7 +82,11 @@ function getNormalizedEndDate(endDate, duration) {
 
     // CalDav saves the end date of all day events as the start of the next day
     // To fix this, we subtract one second from the end date
-    return allDayEvent ? moment(endDate).subtract(1, "seconds").toISOString() : endDate;
+    return (
+        allDayEvent ?
+            new Date(moment(endDate).subtract(1, "seconds")) :
```

Comment:
You don't have to convert it to `new Date()`. The moment object has a `toISOString()` method as well.